server/codecity.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * @license
+ * Code City: Server
+ *
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview A virtual world of collaborative coding.
+ * @author fraser@google.com (Neil Fraser)
+ */
+
+// Start with: node codecity.js <DB directory>
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const Interpreter = require('./');
+const Serializer = require('./serialize.js');
+
+var databaseDirectory;
+var interpreter;
+
+function startup() {
+  // process.argv is a list containing: ['node', 'codecity.js', 'databaseDir']
+  databaseDirectory = process.argv[2];
+
+  // Check that the directory was specified and exists.
+  try {
+    var files = fs.readdirSync(databaseDirectory);
+  } catch (e) {
+    console.log('Database not found.\nUsage: node ' + process.argv[1] +
+                ' <DB directory>');
+    process.exit(1);
+  }
+
+  // Find the most recent database file.
+  files.sort();
+  for (var i = files.length - 1; i >= 0; i--) {
+    if (files[i].match(/^\d\d\d\d-\d\d-\d\dT\d\d-\d\d-\d\d(\.\d)?\d?\d?Z?\.city$/)) {
```

Comment:
Added {4} and {1-3} to the regex, though {2} is longer than what it replaces.
Don't have any strong opinions regarding the time separator (other than disliking :, / or nothing).